krod/modules/research/document_processor.py

```
@@ -165,16 +173,21 @@ async def validate_evidence(
-        citations = ["**Sources:**"]
-        for i, src in enumerate(sources, 1):
-            confidence = f" (Confidence: {src.confidence:.0%})" if hasattr(src, 'confidence') else ""
-            citations.append(
-                f"{i}. {src.to_citation()}{confidence}"
-                f"\n   Source: {src.url}"
-                f"\n   Extract: {getattr(src, 'extract', '')[:150]}..."
-            )
-            
-        return "\n".join(citations)
\ No newline at end of file
+        if not hasattr(self, 'sources') or not self.sources:
```

Pylint (3.3.7)

[error] Instance of 'DocumentProcessor' has no 'sources' member

(E1101)